### PR TITLE
Search bar

### PR DESCRIPTION
**Pull request by cbuchko**
Added a search bar to the View Schedules page. Currently it just filters the calendar view but it should work for the table view once that's hooked up.

### CODE REVIEW

File: src/components/Schedule/SearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        let appointmentValues = "";
+        Object.values(appointment).forEach(
+          (value) => (appointmentValues += value.toLowerCase())
+        );
+        for (const word of input) {
+          if (!appointmentValues.includes(word)) {
+            return false;
+          }
+        }
+        return true;
```
Comment (jayden-chan) [inline src/components/Schedule/SearchBar.tsx]:
```suggestion
	return input.every((word) => appointmentValues.includes(word));
```

File: src/components/Schedule/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  const filterCourses = (clear?: boolean) => {
+    if (searchInput === "" || clear) {
+      setFiltered("");
+      return setScheduleData(termData);
+    }
+
+    //checks if every word of the search exists as an attribute for the course
+    const input = searchInput.toLowerCase().split(" ");
+    const filteredAppointments = termData.filter(
```
Comment (jayden-chan) [inline src/components/Schedule/SearchBar.tsx]: Might want to rename this from `input` to `inputWords` or something more descriptive

**Review by jayden-chan — APPROVED** [review]
Looks good, no blockers just some minor tweaks. I didn't build and test this at all so I'm just having blind faith that it all works :stuck_out_tongue:

File: src/components/Schedule/SearchBar.tsx (hunk 1 above)
Comment (cbuchko) [reply src/components/Schedule/SearchBar.tsx]: god you are smart

File: src/components/Schedule/SearchBar.tsx (hunk 2 above)
Comment (cbuchko) [reply src/components/Schedule/SearchBar.tsx]: updated

**Review by Adamk8 — APPROVED** [review]
I tested it locally and it seems t be working good!

**Pull request merged**